packages/client-config/src/client-config-contributor/custom_client_config_contributor.ts

```diff
@@ -0,0 +1,68 @@
+import { ClientConfigContributor } from './client_config_contributor.js';
+import { UnifiedBackendOutput } from '@aws-amplify/backend-output-schemas';
+import { CustomClientConfig } from '../client-config-types/client_config.js';
+
+// TODO: these types are copied, should be moved to platform-types perhaps.
+export type ClientConfigDestination = {
+  clientConfigFormat: string;
+  path: Array<string>;
+}
+
+export type CustomOutputOptions = {
+  clientConfigDestinations?: Array<ClientConfigDestination>
+}
+
+export class CustomClientConfigContributor implements ClientConfigContributor {
+  contribute = async ({
+    ['AWS::Amplify::Custom']: customOutputs,
+  }: UnifiedBackendOutput): Promise<
+    CustomClientConfig | Record<string, never>
+  > => {
+    if (
+      customOutputs === undefined ||
+      Object.keys(customOutputs.payload).length === 0
+    ) {
+      console.log('custom outputs undefined');
+      return {};
+    }
+
+    console.log(customOutputs);
+
+    const result: CustomClientConfig & Record<string, unknown> = { custom: {} };
+
+    Object.entries(customOutputs.payload).forEach(([key, value]) => {
+      const customOutputData = JSON.parse(value);
+      // TODO how do we do custom namespacing ??
+      result.custom[key.replace('amplifycustom', '')] = customOutputData.value;
+
+      const customOutputOptions: CustomOutputOptions | undefined = customOutputData.options;
+
+      if (customOutputOptions){
+        // TODO should this be in addition to placing value in custom.X section or should these be XOR ?
+        customOutputOptions.clientConfigDestinations?.forEach(clientConfigDestination => {
+          let currentRef: Record<string, unknown> = result;
+          let currentArrayRef: Array<unknown> | undefined = undefined;
+          for (let i = 0 ; i < clientConfigDestination.path.length - 1; i++) {
+            const pathSegment = clientConfigDestination.path[i];
+            if (!currentRef[pathSegment]) {
+              currentRef[pathSegment] = {}
+            }
+            currentRef = currentRef[pathSegment] as Record<string, unknown>;
+          }
+          const lastPathSegment = clientConfigDestination.path[clientConfigDestination.path.length - 1];
+          // TODO this is a hack
```

Comment:
recent CVE not withstanding, [lodash set](https://lodash.com/docs/4.17.15#set) could be the right tool for the job here